packages/my-queries-storage/src/provider.ts

```diff
@@ -13,38 +14,51 @@ export type RecentQueryStorageAccess = {
   getStorage(options?: QueryStorageOptions): RecentQueryStorage;
 };
 
-const PipelineStorageContext = createContext<PipelineStorage | null>(null);
-const FavoriteQueryStorageContext =
-  createContext<FavoriteQueryStorageAccess | null>(null);
-const RecentQueryStorageContext =
-  createContext<RecentQueryStorageAccess | null>(null);
+const PipelineStorageContext = createContext<PipelineStorage | undefined>(
+  undefined
+);
+const FavoriteQueryStorageContext = createContext<
+  FavoriteQueryStorageAccess | undefined
+>(undefined);
+const RecentQueryStorageContext = createContext<
+  RecentQueryStorageAccess | undefined
+>(undefined);
 
 export const PipelineStorageProvider = PipelineStorageContext.Provider;
 export const FavoriteQueryStorageProvider =
   FavoriteQueryStorageContext.Provider;
 export const RecentQueryStorageProvider = RecentQueryStorageContext.Provider;
 
-export const pipelineStorageLocator = (): PipelineStorage => {
+export const pipelineStorageLocator = (): PipelineStorage | undefined => {
+  const storageIsRequired = usePreference('enableSavedAggregationsQueries');
   const pipelineStorage = useContext(PipelineStorageContext);
-  if (!pipelineStorage) {
+  if (storageIsRequired && !pipelineStorage) {
     throw new Error('No PipelineStorage available in this context');
   }
+
   return pipelineStorage;
 };
 
-export const favoriteQueryStorageAccessLocator =
-  (): FavoriteQueryStorageAccess => {
-    const favoriteQueryStorageAccess = useContext(FavoriteQueryStorageContext);
-    if (!favoriteQueryStorageAccess) {
-      throw new Error('No FavoriteQueryStorage available in this context');
-    }
-    return favoriteQueryStorageAccess;
-  };
-
-export const recentQueryStorageAccessLocator = (): RecentQueryStorageAccess => {
-  const recentQueryStorageAccess = useContext(RecentQueryStorageContext);
-  if (!recentQueryStorageAccess) {
-    throw new Error('No RecentQueryStorage available in this context');
+export const favoriteQueryStorageAccessLocator = ():
+  | FavoriteQueryStorageAccess
+  | undefined => {
+  const storageIsRequired = usePreference('enableSavedAggregationsQueries');
```

Comment:
One way to use the preferences was this, to use the hook and assume that we are in a tree below the PreferencesProvider. The other way was to provider preferencesInstance alongside these storage values in the providers so that we can access that instance here in locator, similar to how this is done for LoggerAndTelemetryProvider.

I opted for this one only because it will keep the CompassWeb render tree free of these Providers, otherwise we will have to have them there as well passing just the preferences in. LMK if its alright with you all.

Reply:
Summarising my discussion with @addaleax:
We can simplify this peace here by not having this preference altogether because at this point this preference is not giving much value on top of the presence/absence of storage values in providers and we would be better off by just checking the presence of the storage in the provider and doing the same that we were earlier doing with the preference.

Reply:
Addressed this in [5325252](https://github.com/mongodb-js/compass/pull/5454/commits/532525211d7723217019504e3525e41dc75056f3)